homeassistant/components/xiaomi_miio/button.py

```diff
@@ -0,0 +1,126 @@
+"""Support for Xiaomi buttons."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    CONF_MODEL,
+    DOMAIN,
+    KEY_COORDINATOR,
+    KEY_DEVICE,
+    MODEL_AIRFRESH_A1,
+    MODEL_AIRFRESH_T2017,
+)
+from .device import XiaomiCoordinatedMiioEntity
+
+DATA_KEY = "button.xiaomi_miio"
+
+ATTR_RESET_DUST_FILTER = "reset_dust_filter"
+ATTR_RESET_UPPER_FILTER = "reset_upper_filter"
+
+
+@dataclass
+class XiaomiMiioButtonDescription(ButtonEntityDescription):
+    """A class that describes button entities."""
+
+    method_press: str = ""
+    method_press_error_message: str = ""
+
+
+BUTTON_TYPES = (
+    XiaomiMiioButtonDescription(
+        key=ATTR_RESET_DUST_FILTER,
+        name="Reset Dust Filter",
+        icon="mdi:air-filter",
+        method_press="reset_dust_filter",
+        method_press_error_message="Resetting the dust filter lifetime failed",
+        entity_category=EntityCategory.CONFIG,
+    ),
+    XiaomiMiioButtonDescription(
+        key=ATTR_RESET_UPPER_FILTER,
+        name="Reset Upper Filter",
+        icon="mdi:air-filter",
+        method_press="reset_upper_filter",
+        method_press_error_message="Resetting the dust filter lifetime failed.",
+        entity_category=EntityCategory.CONFIG,
+    ),
+)
+
+MODEL_TO_BUTTON_MAP = {
+    MODEL_AIRFRESH_A1: (ATTR_RESET_DUST_FILTER,),
+    MODEL_AIRFRESH_T2017: (
+        ATTR_RESET_DUST_FILTER,
+        ATTR_RESET_UPPER_FILTER,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the button from a config entry."""
+    entities = []
```

Comment:
Please move the definitions that we don't need until later down after the check of `MODEL_TO_BUTTON_MAP`.